include/revng/ABI/CallingConventionTrait.h

```diff
@@ -0,0 +1,1083 @@
+  /// \note: If `MaximumGPRsPerScalarArgument` is equal to 0, it means that
+  /// the ABI does not allow scalar arguments to use GPRs.
+  ///
+  /// \note: If an argument doesn't fit into the specified registers or
+  /// uses irregular padding, the registers are not used and the object is
+  /// passed using the memory (stack, pointer-to-copy, etc.).
+  static constexpr size_t MaximumGPRsPerScalarArgument = 0;
+
+  /// States the maximum number of GPRs available to return a single scalar
+  /// (`int`, `__int128`, pointer, etc.) value, meaning that it can only be
+  /// returned in the GPRs if `MaximumGPRsPerScalarReturnValue` is less than
+  /// or equal to the number of the registers required to fit the object
+  /// including padding.
+  ///
+  /// \note: If `MaximumGPRsPerScalarReturnValue` is equal to 0, it means
+  /// that the ABI does not allow scalar return values to use GPRs.
+  ///
+  /// \note: If a return value doesn't fit into the specified registers or
+  /// uses irregular padding, the registers are not used and the object is
+  /// passed using the memory (stack, pointer-to-copy, etc.).
+  static constexpr size_t MaximumGPRsPerScalarReturnValue = 0;
+
+  /// States the required size for a single stack argument in bytes.
+  static constexpr size_t MinimumStackArgumentSize = 1;
+
+  /// Stores the list of general purpose registers allowed to be used for
+  /// passing arguments and the order they are to be used in.
+  static constexpr std::array<model::Register::Values, 0>
+    GeneralPurposeArgumentRegisters = {};
+
+  /// Stores the list of general purpose registers allowed to be used for
+  /// returning values and the order they are to be used in.
+  static constexpr std::array<model::Register::Values, 0>
+    GeneralPurposeReturnValueRegisters = {};
+
+  /// Stores the list of vector registers allowed to be used for passing
+  /// arguments and the order they are to be used in.
+  static constexpr std::array<model::Register::Values, 0>
+    VectorArgumentRegisters = {};
+
+  /// Stores the list of vector registers allowed to be used for returning
+  /// values and the order they are to be used in.
+  static constexpr std::array<model::Register::Values, 0>
+    VectorReturnValueRegisters = {};
+
+  /// Stores the list of registers for which the ABI requires the callee to
+  /// preserve the value, meaning that when the callee returns, the value of
+  /// those registers must be the same as it was when the function was called.
+  static constexpr std::array<model::Register::Values, 0>
+    CalleeSavedRegisters = {};
+
+  /// Specifies a register to be used for returning (or even passing,
+  /// depending on ABI) the pointer to the memory used for returning
+  /// copies of big aggregate objects.
+  ///
+  /// Can be `model::Register::Invalid` for ABIs that do not support returning
+  /// values by 'pointer-to-copy'.
+  static constexpr auto ReturnValueLocationRegister = model::Register::Invalid;
+};
+
+/// System V x64 ABI.
```

Comment:
Let's have one file per ABI, no reason to have such a gigantic file.

Reply:
[Done](https://github.com/revng/revng/pull/98/commits/8268d59c39561de3d5bbc17397dda3983a816121).